README.md

```diff
@@ -1,76 +1,207 @@
 
-To run one of the benchmarks:
+We're learning together here.
 
-1. `cd` into desired benchmark directory (EG `$ cd loops`)
-2. Compile by running `$ ../compile.sh`
-3. Run via `$ ../run.sh`.
-  You should see output something like:
-  
-  ```
-  $ ../run.sh
+* If you have a suggestion for improvement, want to add a language, a benchmark, fix a bug or typo: Issues and PRs. Which one to use depends. But generally is is first Issue, then PR, where you can use the issue to formulate the problem statement, and PRs to address the problem. Use your judgement and we will succeed.
+* Have a question? -> Issue.
+
+## Running the benchmarks
+
+To run benchmarks you need toolchains to run (and often to compile) the programs for the languages you want to benchmark. The scripts are written so that benchmarks are compiled and run for any language for which you have a working toolchain.
+
+The steps are performed in a per-benchmark fashion by doing `cd` to the benchmark directory and then:
+
+1. Compile the programs that need compiling:
+
+   ```
+   $ ./compile.sh
+   ```
+1. Run, providing your GitHub user handle, e.g.:
+
+   ```
+   $ ./run.sh -u PEZ
+   ```
+
+   (This is what we refer to as [The Runner](#the-runner))
+1. Clean build files:
+
+   ```
+   $ ./clean.sh
+   ```
+
+## The Runner
+
+The general strategy for the runner to benchmark only the targeted function is that it is the programs being benchmarked that do the benchmarking in-process. They only measure around the single piece of work that the benchmark is about. So for **fibonacci** only the call to the function calculating `fibonacci(n)` should be measured. For **levenshtein** benchmark a function that collects all pairing distances is measured. This is because we use the sum of the distances for [correctness check](#correctness-check).
+
+Each program (language) will be allowed the same amount of time to complete the benchmark work (as many times as it can).
+
+Because of the above, each language will have to have some minimal utility/tooling for running the function-under-benchmark as many times as a timeout allows, plus reporting the measurements and the result. Here are three implementations, that we can regard as being reference:
+
+* [benchmark.clj](lib/clojure/src/languages/benchmark.clj)
+* [benchmark.java](lib/java/languages/Benchmark.java)
+* [benchmark.c](lib/c/benchmark.c) (This one may need some scrutiny from C experts before we fully label it as *reference*.)
+
+You'll see that the `benchmark/run` function takes two arguments:
+
+1. `f`: A function (a thunk)
+1. `run-ms`: A total time in milliseconds within which the function should be run as many times as possible
```

Comment:
Please make sure you're using a monotonic clock for these measurements (wall clock time can jump around).

I don't know how this actually works in other languages, but in Racket you can provide a time-limit for a body of execution (see [`call-with-time-limit`](https://docs.racket-lang.org/reference/Sandboxed_Evaluation.html#%28def._%28%28lib._racket%2Fsandbox..rkt%29._call-with-deep-time-limit%29%29)).

Reply:
Good point. I made sure it was a monotonic clock for Java and Clojure, but I shall now check the other language implementations. It's probably fine, but I didn't pay attention to that so could have slipped in some wall clocking.